pymeasure/instruments/agilent/agilentN9320A.py

```diff
@@ -0,0 +1,262 @@
+    D_FACTOR = 2               #A multiplicative factor for the sweep time in
+                               #peak function
+    DELAY = 0.2                #A delay in second can be useful
+
+
+    start_frequency = Instrument.control(
+        ":SENS:FREQ:STAR?", ":SENS:FREQ:STAR %e Hz",
+        """ A floating point property that represents the start frequency
+        in Hz. This property can be set.
+        """,
+		validator=strict_range,
+		values=FREQ_LIMIT
+    )
+    stop_frequency = Instrument.control(
+        ":SENS:FREQ:STOP?", ":SENS:FREQ:STOP %e Hz",
+        """ A floating point property that represents the stop frequency
+        in Hz. This property can be set.
+        """,
+		validator=strict_range,
+		values=FREQ_LIMIT
+    )
+    center_frequency = Instrument.control(
+        ":SENS:FREQ:CENT?", ":SENS:FREQ:CENT %eHz",
+        """ A floating point property that represents the center frequency
+        in Hz. This property can be set.
+        """,
+		validator=strict_range,
+		values=FREQ_LIMIT
+    )
+    span = Instrument.control(
+        ":SENS:FREQ:SPAN?", ":SENS:FREQ:SPAN %eHz",
+        """ A floating point property that represents the frequency span
+        in Hz. This property can be set.
+        """,
+		validator=strict_range,
+		values=SPAN_LIMIT
+    )
+    sweep_time = Instrument.control(
+        ":SENS:SWE:TIME?", ":SENS:SWE:TIME %.2e",
+        """ A floating point property that represents the sweep time
+        in seconds. This property can be set.
+        """
+    )
+    ref_level = Instrument.control(
+        ":DISP:WIND:TRAC:Y:SCAL:RLEV?", ":DISP:WIND:TRAC:Y:SCAL:RLEV %.2f",
+        """ A floating point property that represents the reference level
+        in dBm. This property can be set.
+        """,
+        validator=strict_range,
+        values=REF_LIMIT
+    )
+    att = Instrument.control(
+        ":SENS:POW:RF:ATT?", ":SENS:POW:RF:ATT %d",
+        """ A floating point property that represents the input attenuator
+        in dB. This property can be set.
+        """,
+        validator=strict_range,
+        values=ATT_LIMIT
+    )
+    res_bw = Instrument.control(
+        ":SENS:BAND:RES?", ":SENS:BAND:RES %e",
+        """ A floating point property that represents the resolution
+        bandwidth in Hz. This property can be set.
+        """,
+        validator=strict_discrete_set,
+        values=RES_LIMIT
+    )
+    vid_bw = Instrument.control(
+        ":SENS:BAND:VID?", ":SENS:BAND:VID %e",
+        """ A floating point property that represents the video
+        bandwidth in Hz. This property can be set.
+        """,
+        validator=strict_discrete_set,
+        values=VID_LIMIT
+    )
+
+    def __init__(self, resourceName, **kwargs):
+        super(AgilentN9320A, self).__init__(
+            resourceName,
+            "Agilent AgilentN9320A Spectrum Analyzer",
+            **kwargs
+        )
+    def init_imm(self):
+        self.write("INIT:IMM")
```

Comment:
Please add docstrings to each method. Also consider making the names more human readable, along the same lines as the comments above.